app/controllers/journeys/capitalallowances/zeroEmissionGoodsVehicle/ZeroEmissionGoodsVehicleController.scala

```diff
@@ -0,0 +1,93 @@
+import queries.Settable
+import services.SelfEmploymentService
+import services.SelfEmploymentService.clearDataFromUserAnswers
+import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendBaseController
+import utils.Logging
+import views.html.journeys.capitalallowances.zeroEmissionGoodsVehicle.ZeroEmissionGoodsVehiclesView
+
+import javax.inject.{Inject, Singleton}
+import scala.concurrent.{ExecutionContext, Future}
+
+@Singleton
+class ZeroEmissionGoodsVehicleController @Inject() (override val messagesApi: MessagesApi,
+                                                    navigator: CapitalAllowancesNavigator,
+                                                    identify: IdentifierAction,
+                                                    getData: DataRetrievalAction,
+                                                    requireData: DataRequiredAction,
+                                                    service: SelfEmploymentService,
+                                                    formProvider: ZeroEmissionGoodsVehicleFormProvider,
+                                                    val controllerComponents: MessagesControllerComponents,
+                                                    view: ZeroEmissionGoodsVehiclesView)(implicit ec: ExecutionContext)
+    extends FrontendBaseController
+    with I18nSupport
+    with Logging {
+
+  def onPageLoad(taxYear: TaxYear, businessId: BusinessId, mode: Mode): Action[AnyContent] = (identify andThen getData andThen requireData) {
+    implicit request =>
+      val form = request.userAnswers
+        .get(ZeroEmissionGoodsVehiclePage, businessId.some)
+        .fold(formProvider(request.userType, taxYear))(formProvider(request.userType, taxYear).fill)
+
+      Ok(view(form, mode, request.userType, taxYear, businessId))
+  }
+
+  def onSubmit(taxYear: TaxYear, businessId: BusinessId, mode: Mode): Action[AnyContent] = (identify andThen getData andThen requireData) async {
+    implicit request =>
+      formProvider(request.userType, taxYear)
+        .bindFromRequest()
+        .fold(
+          formErrors => Future.successful(BadRequest(view(formErrors, mode, request.userType, taxYear, businessId))),
+          answer =>
+            for {
+              (editedUserAnswers, redirectMode) <- handleGatewayQuestion(answer, request, mode, businessId)
+              updatedUserAnswers                <- service.persistAnswer(businessId, editedUserAnswers, answer, ZeroEmissionGoodsVehiclePage)
+            } yield Redirect(navigator.nextPage(ZeroEmissionGoodsVehiclePage, redirectMode, updatedUserAnswers, taxYear, businessId))
+        )
+  }
+
+  private def handleGatewayQuestion(currentAnswer: Boolean,
+                                    request: DataRequest[_],
+                                    mode: Mode,
+                                    businessId: BusinessId): Future[(UserAnswers, Mode)] = {
+    val pagesToBeCleared: List[Settable[_]] = List()
```

Comment:
More idiomatic way to create empty list is `= Nil`

Reply:
Leaving an empty list to remember to add each page into the list as each page is created

Reply:
ok